### PR TITLE
bug(query): step cannot be 0 error for instant query

### PR DESCRIPTION
**Pull request by TanviBhavsar**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** 
Absent and topk instant queries throw `step cannot be 0` error

**New behavior :**
Change for loop to use Iterator

### CODE REVIEW

File: query/src/main/scala/filodb/query/exec/RangeVectorTransformer.scala [hunk 1]
```diff
@@ -368,10 +369,16 @@ final case class AbsentFunctionMapper(columnFilter: Seq[ColumnFilter], rangePara
...
+
+      val it = Iterator.from(0, rangeParams.stepSecs.toInt)
+        .takeWhile(_ <= rangeParams.endSecs - rangeParams.startSecs).map { i =>
+        val timestamp = i + rangeParams.startSecs
+        if (!t.contains(timestamp * 1000))
+          rowList += new TransientRow(timestamp * 1000, 1)
+       }
+
+      // address step == 0 case
+      if (rangeParams.startSecs == rangeParams.endSecs) it.take(1).toList else it.toList
```
Comment (sherali42) [inline query/src/main/scala/filodb/query/exec/RangeVectorTransformer.scala:381]: potential infinite loop here, can you add `step == 0` condition also

File: query/src/main/scala/filodb/query/exec/aggregator/TopBottomKRowAggregator.scala [hunk 2]
```diff
@@ -100,9 +100,13 @@ class TopBottomKRowAggregator(k: Int, bottomK: Boolean) extends RowAggregator {
...
   private def createBuilder(rangeParams: RangeParams, currentTime: Long): RecordBuilder= {
     val builder = SerializedRangeVector.newBuilder();
-    for (t <- rangeParams.startSecs to (currentTime - rangeParams.stepSecs) by rangeParams.stepSecs) {
+    val it = Iterator.from(0, rangeParams.stepSecs.toInt)
+      .takeWhile(_ <= (currentTime - rangeParams.stepSecs) - rangeParams.startSecs).map { i =>
+      val t = i + rangeParams.startSecs
       addRecordToBuilder(builder, t * 1000, Double.NaN)
     }
+    // when step == 0 we don't need to add to builder
+    if (rangeParams.startSecs != rangeParams.endSecs) it.toList
```
Comment (sherali42) [inline query/src/main/scala/filodb/query/exec/aggregator/TopBottomKRowAggregator.scala:109]: potential infinite loop here, can you add `step == 0` condition also

File: query/src/main/scala/filodb/query/exec/aggregator/TopBottomKRowAggregator.scala [hunk 3]
```diff
@@ -113,27 +117,32 @@ class TopBottomKRowAggregator(k: Int, bottomK: Boolean) extends RowAggregator {
...
           }
           i += 2
         }
         resRvs.keySet.foreach { rvs =>
-          if (!rvkSeen.contains(rvs)) addRecordToBuilder(resRvs.get(rvs).get, t * 1000, Double.NaN)
+          if (!rvkSeen.contains(rvs)) addRecordToBuilder(resRvs.get(rvs).get, timestamp * 1000, Double.NaN)
         }
       }
+      // address step == 0 case
+      if (rangeParams.startSecs == rangeParams.endSecs) it.take(1).toList else it.toList
```
Comment (sherali42) [inline query/src/main/scala/filodb/query/exec/aggregator/TopBottomKRowAggregator.scala:140]: can you add `step == 0` condition also

**Review by sherali42 — APPROVED** [review]
one minor comment, otherwise looks good

File: query/src/main/scala/filodb/query/exec/RangeVectorTransformer.scala (hunk 1 above)
Comment (TanviBhavsar) [reply query/src/main/scala/filodb/query/exec/RangeVectorTransformer.scala]: Done

File: query/src/main/scala/filodb/query/exec/aggregator/TopBottomKRowAggregator.scala (hunk 2 above)
Comment (TanviBhavsar) [reply query/src/main/scala/filodb/query/exec/aggregator/TopBottomKRowAggregator.scala:109]: Not needed here

File: query/src/main/scala/filodb/query/exec/aggregator/TopBottomKRowAggregator.scala (hunk 3 above)
Comment (TanviBhavsar) [reply query/src/main/scala/filodb/query/exec/aggregator/TopBottomKRowAggregator.scala]: Done

**Review by sherali42 — APPROVED** [review]
thanks for the change @TanviBhavsar, 👍

**Pull request merged**